packages/@tinacms/graphql/src/database/index.ts

```diff
@@ -0,0 +1,537 @@
+          }
+        }
+        return false
+      })
+
+      let data = contentObject
+      if ((extension === '.md' || extension === '.mdx') && field) {
+        if (hasOwnProperty(contentObject, '$_body')) {
+          const { $_body, ...rest } = contentObject
+          data = rest
+          data[field.name] = $_body
+        }
+      }
+      return {
+        ...data,
+        _collection: collection.name,
+        _keepTemplateKey: !!collection.templates,
+        _template: lastItem(template.namespace),
+        _relativePath: filepath
+          .replace(collection.path, '')
+          .replace(/^\/|\/$/g, ''),
+        _id: filepath,
+      } as T
+    }
+  }
+
+  public addPendingDocument = async (
+    filepath: string,
+    data: { [key: string]: unknown }
+  ) => {
+    const { stringifiedFile, payload } = await this.stringifyFile(
+      filepath,
+      data
+    )
+    if (this.store.supportsSeeding()) {
+      await this.bridge.put(filepath, stringifiedFile)
+    }
+    await this.store.put(filepath, payload)
+  }
+
+  public put = async (filepath: string, data: { [key: string]: unknown }) => {
+    if (SYSTEM_FILES.includes(filepath)) {
+      throw new Error(`Unexpected put for config file ${filepath}`)
+    } else {
+      const { stringifiedFile, payload } = await this.stringifyFile(
+        filepath,
+        data
+      )
+      if (this.store.supportsSeeding()) {
+        await this.bridge.put(filepath, stringifiedFile)
+      }
+      const existingData = await this.get<{ _collection: string }>(filepath)
+      const collection = this.tinaSchema.getCollection(existingData._collection)
+      /**
+       * Determine the attributes which are already in the store and
+       * just remove them all. This can be more efficient if we only
+       * remove what's no longer needed, but keeping it simple for now.
+       */
+      const attributesToFilterOut = await _indexCollectable({
+        record: filepath,
+        value: existingData,
+        field: collection,
+        prefix: collection.name,
+        database: this,
+      })
+      /**
+       * As a separate step, add the "new" attributes, even though
+       * they'll often be the same as the ones we just removed
+       */
+      const attributesToAdd = await _indexCollectable({
+        record: filepath,
+        value: data,
+        field: collection,
+        prefix: collection.name,
+        database: this,
+      })
+      await sequential(attributesToFilterOut, async (attribute) => {
+        const records = (await this.store.get<string[]>(attribute)) || []
+        await this.store.put(
+          attribute,
+          records.filter((item) => item !== filepath)
+        )
+        return true
+      })
+      await sequential(attributesToAdd, async (attribute) => {
+        const records = (await this.store.get<string[]>(attribute)) || []
+        await this.store.put(attribute, [...records, filepath])
+        return true
+      })
```

Comment:
This goes through and removes the 'metadata' attributes, and re-adds the new ones.